zone/tradeskills.cpp

```diff
@@ -1866,37 +1866,53 @@ bool ZoneDatabase::DisableRecipe(uint32 recipe_id)
 bool Client::CheckTradeskillLoreConflict(int32 recipe_id)
 {
 	auto recipe_entries = TradeskillRecipeEntriesRepository::GetWhere(
-			content_db,
-			fmt::format(
-					"recipe_id = {} ORDER BY componentcount DESC",
-					recipe_id
-			)
+		content_db,
+		fmt::format(
+			"recipe_id = {} ORDER BY componentcount DESC",
+			recipe_id
+		)
 	);
 	if (recipe_entries.empty()) {
 		return false;
 	}
 
-	for (auto& f : recipe_entries) {
-		for (auto &e: recipe_entries) {
-			if (f.componentcount > 0 && e.item_id == f.item_id && e.componentcount == 0) {
-				e.item_id = 0;
+	const EQ::ItemData* fi = nullptr;
+	const EQ::ItemData* ei = nullptr;
+	for (auto &f : recipe_entries) {
+		if (f.item_id) {
+			fi = database.GetItem(f.item_id);
+			for (auto &e : recipe_entries) {
+				bool fi_is_valid = e.item_id && fi && fi->LoreGroup != 0;
+				bool item_id_match = e.item_id == f.item_id;
```

Comment:
This may not necessarily need a bool to describe intent, it's pretty clear to the next traveler that this check is looking for where item_id's are equal

Reply:
makes sense to me, removed it.